src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.SectionNameMatching.cs

```diff
@@ -114,6 +114,26 @@ public bool IsMatch(string s)
                 numberRangePairs.ToImmutableAndFree());
         }
 
+        internal static bool TryUnescapeSectionName(string sectionName, out string? escapedSectionName)
+        {
+            var sb = new StringBuilder();
+            SectionNameLexer lexer = new SectionNameLexer(sectionName);
+            while (!lexer.IsDone)
+            {
+                var tokenKind = lexer.Lex();
+                if (tokenKind == TokenKind.SimpleCharacter)
+                {
+                    sb.Append(lexer.EatCurrentCharacter());
+                }
+                else
+                {
+                    sb.Append(lexer.CurrentCharacter);
```

Comment:
>sb.Append(lexer.CurrentCharacter) [](http://example.com/codeflow?start=20&length=33)

Do we ever hit this code path? I may be misreading the `Lex()` method, but it looks like `CurrentCharacter` is the character after the special character in this case. Is that expected here?

Reply:
Yeah, you're right. This path never gets hit. I'm not sure why I had this in the first place (perhaps something came up when I was doing the end-to-end testing with the dotnet/website repo), but it's not needed. `CanGetSectionsWithSpecialCharacters` still passes without this.